pdr_backend/pdr_dashboard/callbacks/callbacks_feeds.py

```diff
@@ -0,0 +1,254 @@
+import dash
+from dash import Input, Output, State, callback_context
+from pdr_backend.pdr_dashboard.pages.feeds import FeedsPage
+from pdr_backend.pdr_dashboard.dash_components.plots import (
+    get_feed_figures,
+    FeedModalFigures,
+)
+from pdr_backend.cli.arg_feeds import ArgFeed
+from pdr_backend.pdr_dashboard.util.format import format_table
+from pdr_backend.pdr_dashboard.util.data import get_feed_column_ids
+
+
+def filter_table_by_range(min_val, max_val, label_text):
+    ctx = callback_context
+    button_id = ctx.triggered[0]["prop_id"].split(".")[0]
+    if (not min_val and not max_val) or button_id == "clear_filters_button":
+        return label_text
+
+    return f"{label_text} {min_val if min_val else ''}-{max_val if max_val else ''}"
+
+
+def filter_condition(item, field, values):
+    return not values or item[field] in values
+
+
+def search_condition(item, search_value):
+    if not search_value:
+        return True
+    search_value = search_value.lower()
+    return any(
+        search_value in item.get(key, "").lower()
+        for key in ["addr", "base_token", "quote_token"]
+    )
+
+
+def range_condition(item, field, min_value, max_value):
+    item_value = float(item[field])
+    return not (
+        (min_value is not None and min_value != "" and item_value < min_value)
+        or (max_value is not None and max_value != "" and item_value > max_value)
+    )
+
+
+def check_condition(item, condition_type, field, *values):
+    if condition_type == "search":
+        return search_condition(item, values[0])
+    if condition_type == "filter":
+        return filter_condition(item, field, values[0])
+    if condition_type == "range":
+        return range_condition(
+            item,
+            field,
+            float(values[0]) if values[0] is not None and values[0] != "" else None,
```

Comment:
the complexity around this part is due to lack of typing. Settle the type of the function and then we only do conversions once. Settle on a single "empty"-like value, either None or "".